specs/phase1/shard-fork-choice.md

```diff
@@ -59,7 +59,9 @@ def get_shard_latest_attesting_balance(store: Store, shard_store: ShardStore, ro
     return Gwei(sum(
         state.validators[i].effective_balance for i in active_indices
         if (
-            i in store.latest_messages and get_shard_ancestor(
+            i in store.latest_messages and
+            store.latest_messages[i].shard == shard_store.shard and
```

Comment:
I don't understand this. It seems that only if the validator has attested to the shard _in their latest message_ that their vote would be counted. So in the next slot, when the attest to a different shard, their vote would be removed. Is that how it should work?

Reply:
It would be in the next epoch rather than slot.
The beacon chain will generally continue to advance to provide a new basis for the shard, and if not, committees will continue to turn over and vote on the shard.

Even if we store multiple shard votes for a single validator, this condition is still important. We need to prevent using a vote on shard A to try to influence shard B (example: using your assigned vote from shard A to vote on a root from shard B, thus throwing your weight behind a shard you weren't assigned)

Reply:
True but even if it's in the last epoch, it means the "old" blocks suddenly lose their attestations -- so the commitee in the next epoch could just ignore them if they haven't been crosslinked. Seems weird to me.

Reply:
How about we flag this for discussion on Tuesday? I think it's worth hashing out in convo